contracts/peripheral/YieldDirectorGohm.sol

```diff
@@ -0,0 +1,363 @@
+	            addresses[index] = currDeposit.recipient;
+	            agnosticDeposits[index] = IgOHM(gOHM).balanceTo(currDeposit.principalAmount);
+	        }
+        }
+
+        return (addresses, agnosticDeposits);
+    }
+
+    /**
+        @notice Return total amount of gOHM donated to recipient since last full withdrawal or redemption
+        @param donor_ Address of user donating yield
+        @param recipient_ Address of user recieiving donated yield
+     */
+    function donatedTo(address donor_, address recipient_) external view returns (uint256) {
+        uint256[] storage depositIds = depositorIds[donor_];
+
+        for (uint256 index = 0; index < depositIds.length; index++) {
+            DepositInfo storage currDeposit = depositInfo[depositIds[index]];
+            if (currDeposit.recipient == recipient_) {
+                return _getOutstandingYield(currDeposit.principalAmount, currDeposit.agnosticAmount);
+            }
+        }
+
+        return 0;
+    }
+
+    /**
+        @notice Return total amount of gOHM donated from donor since last full withdrawal or redemption
+        @param donor_ Address of user donating yield
+     */
+    function totalDonated(address donor_) external view returns (uint256) {
+        uint256[] storage depositIds = depositorIds[donor_];
+        uint256 principalTotal = 0;
+        uint256 agnosticTotal = 0;
+
+        for (uint256 index = 0; index < depositIds.length; index++) {
+            DepositInfo storage currDeposit = depositInfo[depositIds[index]];
+
+            principalTotal += currDeposit.principalAmount;
+            agnosticTotal += currDeposit.agnosticAmount;
+        }
+
+        return _getOutstandingYield(principalTotal, agnosticTotal);
+    }
+
+    /************************
+    * Recipient Functions
+    ************************/
+
+    function redeemableBalance(uint256 id_) public view returns (uint256) {
+        DepositInfo storage currDeposit = depositInfo[id_];
+        return _getOutstandingYield(currDeposit.principalAmount, currDeposit.agnosticAmount);     
+    }
+
+    /**
+        @notice Get redeemable sOHM balance of a recipient address
+        @param recipient_ Address of user receiving donated yield
+     */
+    function totalRedeemableBalance(address recipient_) public view returns ( uint256 ) {
+        uint256[] storage receiptIds = recipientIds[recipient_];
+        uint agnosticRedeemable = 0;
+
+        for (uint256 index = 0; index < receiptIds.length; index++) {
+            DepositInfo storage currDeposit = depositInfo[receiptIds[index]];
+            agnosticRedeemable += _getOutstandingYield(currDeposit.principalAmount, currDeposit.agnosticAmount);
+        }
+
+        return agnosticRedeemable;
+    }
+
+    /**
+        @notice Redeem recipient's donated amount of sOHM at current index from one donor
+        @param id_ Deposit id for this donation
+     */
+    function redeemYield(uint256 id_) external returns (uint256) {
```

Comment:
remember seeing ur message about closing deposit after redeem yield when principal is 0 instead of on withdraw. Is that not done yet?